### PR TITLE
Removing multiple options to instantiate Image class

### PR DESCRIPTION
**Pull request by mukultaneja**
This commit does following,
1. Modifies the Image class to be instantiated
with a generic repotag property
2. Removes the ImageID property
3. Modifies the tests appropriately

Resolved: #747

Signed-off-by: mukultaneja <mtaneja@vmware.com>

### CODE REVIEW

File: tern/classes/image.py [hunk 1]
```diff
@@ -26,9 +26,9 @@ class Image:
         add_checksums: add new checksums in the existing list of the checksums
         to_dict: return a python dictionary representation of the image
     '''
-    def __init__(self, image_id=None):
+    def __init__(self, repotag=None):
         '''Either initialize using id'''
```
Comment (nishakm) [inline tern/classes/image.py:30]: Can you change this to `'''Initialize using the image's repo name and tag string'''`?

File: tern/classes/docker_image.py [hunk 2]
```diff
@@ -24,22 +24,21 @@ class DockerImage(Image):
         history: a list of commands used to create the filesystem layers
         to_dict: return a dict representation of the object
     '''
-    def __init__(self, repotag=None, image_id=None):
+    def __init__(self, repotag=None):
         '''Initialize using repotag and image_id'''
```
Comment (nishakm) [inline tern/classes/docker_image.py:28]: Can you change this to `'''Initialize using repotag'''`?

File: tern/classes/docker_image.py [hunk 3]
```diff
@@ -71,14 +66,10 @@ def to_dict(self, template=None):
...
-        NameError. If both were used return the id'''
-        if self.repotag is not None and self.image_id is not None:
-            return self.image_id
+        NameError.'''
         if self.repotag is not None:
             return self.repotag
-        if self.image_id is not None:
-            return self.image_id
-        raise NameError("Image object initialized with no repotag or ID")
+        raise NameError("Image object initialized with no repotag")
```
Comment (nishakm) [inline tern/classes/docker_image.py:72]: This whole function can be deleted. There is already a check for no repotag. You will then need to make a change in `load_image()`:
```
container.extract_image_metadata(option) --> container.extract_image_metadata(self.repotag)
```

File: tern/classes/image.py (hunk 1 above)
Comment (mukultaneja) [reply tern/classes/image.py:30]: @nishakm shouldn't it be same like `'''Initialize using repotag'''`?

File: tern/classes/docker_image.py [hunk 4]
```diff
@@ -24,35 +24,32 @@ class DockerImage(Image):
         history: a list of commands used to create the filesystem layers
         to_dict: return a dict representation of the object
     '''
-    def __init__(self, repotag=None, image_id=None):
-        '''Initialize using repotag and image_id'''
-        super().__init__(image_id)
-        self.__repotag = repotag
+
```
Comment (nishakm) [inline tern/classes/docker_image.py:27]: Small nit: can you remove this extra line?

**Pull request merged**